packages/orchestration/src/utils/chainHub.js

```diff
@@ -0,0 +1,86 @@
+import { makeScalarBigMapStore } from '@agoric/vat-data';
+import { E } from '@endo/far';
+import { M } from '@endo/patterns';
+import { CosmosChainInfoShape, IBCConnectionInfoShape } from '../typeGuards.js';
+
+/**
+ * @import {NameHub} from '@agoric/vats';
+ * @import {CosmosChainInfo, IBCConnectionInfo} from '../cosmos-api.js';
+ * @import {Remote} from '@agoric/internal';
+ */
+
+// TODO define key hierarchy in shared constants
+/** agoricNames key for ChainInfo hub */
+export const CHAIN_KEY = 'chain';
+/** namehub for connection info */
+export const CONNECTIONS_KEY = 'connections';
+
+/** @param {Remote<NameHub>} agoricNames */
+export const makeChainHub = agoricNames => {
+  // XXX not reachably-durable? really?? take a zone arg?
+  /** @type {MapStore<string, CosmosChainInfo>} */
+  const chainInfos = makeScalarBigMapStore('chainInfos', {
+    keyShape: M.string(),
+    valueShape: CosmosChainInfoShape,
+  });
+  /** @type {MapStore<string, IBCConnectionInfo>} */
+  const connectionInfos = makeScalarBigMapStore('connectionInfos', {
+    keyShape: M.string(),
+    valueShape: IBCConnectionInfoShape,
+  });
+
+  const chainHub = harden({
+    /**
+     * @param {string} name
```

Comment:
```suggestion
    /**
     * Register a new chain. The name will override a name in
     * well known chain names.
     *
     * If a durable zone is not provided, registration will not survive a reincarnation of the vat.
     * Then if the chain is not yet in the well known names at that point, it will have to
     * be registered again. In an unchanged contract `start` the call will
     * happen again naturally.
     *
     * @param {string} name
```